workbench-core/datasets/src/ddbDataSetMetadataPlugin.ts

```diff
@@ -199,6 +217,7 @@ export class DdbDataSetMetadataPlugin implements DataSetMetadataPlugin {
         storageName: dataSet.storageName,
         path: dataSet.path,
         awsAccountId: dataSet.awsAccountId!,
+        region: dataSet.region!,
```

Comment:
is there a reason why dataset interface keeps region as optional but you are guaranteeing it here?

Reply:
I was following the paradigm of the other optional parameters, mainly `awsAccountId`. I will go through and check that leaving them optional doesnt break anything.